tools/DevDiagnostics/DevHome.DevDiagnostics/Helpers/LoaderSnapAssistantTool.cs

```diff
@@ -0,0 +1,193 @@
+using Microsoft.UI.Xaml;
+using Microsoft.Win32;
+
+namespace DevHome.DevDiagnostics.Helpers;
+
+public class LoaderSnapAssistantTool
+{
+    private const string WindowsImageETWProvider = "2cb15d1d-5fc1-11d2-abe1-00a0c911f518"; /*EP_Microsoft-Windows-ImageLoad*/
+    private const uint LoaderSnapsFlag = 0x80; /* ETW_UMGL_LDR_SNAPS_FLAG */
+    private readonly DDInsightsService _insightsService;
+    private readonly IDevHomeService _devHomeService;
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+    private readonly Dictionary<int, string> _loaderSnapFailures = new();
+
+    public LoaderSnapAssistantTool()
+    {
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+        _insightsService = Application.Current.GetService<DDInsightsService>();
+        _devHomeService = Application.Current.GetService<IDevHomeService>();
+        Init();
+    }
+
+    private void Init()
+    {
+        var crashDumpAnalyzerThread = new Thread(() =>
+        {
+            MyETWListener();
+        });
+        crashDumpAnalyzerThread.Name = "LoaderSnapAssistantThread";
+        crashDumpAnalyzerThread.Start();
+
+        _devHomeService.MissingFileProcessLaunchFailure += info =>
+        {
+            if (!IsLoaderSnapLoggingEnabledForImage(info.processName))
+            {
+                _dispatcher.TryEnqueue(() =>
+                {
+                    var insight = new InsightPossibleLoaderIssue();
+                    insight.Title = CommonHelper.GetLocalizedString("InsightProcessExitMissingDependenciesTitle");
+                    insight.Text = string.Format(CultureInfo.CurrentCulture, CommonHelper.GetLocalizedString("InsightProcessExitMissingDependencies"), info.processName, info.pid, info.exitCode);
+                    insight.ImageFileName = info.processName;
+                    _insightsService.AddInsight(insight);
+                });
+            }
+            else
+            {
+                lock (_loaderSnapFailures)
+                {
+                    if (_loaderSnapFailures.TryGetValue(info.pid, out string? loadersnapError))
+                    {
+                        // We received information about this app's loader snap issue. Raise the insight.
+                        _loaderSnapFailures.Remove(info.pid);
+
+                        _dispatcher.TryEnqueue(() =>
+                        {
+                            var insight = new SimpleTextInsight();
+                            insight.Title = string.Format(CultureInfo.CurrentCulture, CommonHelper.GetLocalizedString("InsightProcessExitMissingDependenciesIdentifiedTitle"), info.processName, info.pid);
+                            insight.Description = loadersnapError;
+                            _insightsService.AddInsight(insight);
+                        });
+                    }
+                }
+            }
+        };
+    }
+
+    private void MyETWListener()
```

Comment:
Your use of "MyXXX" makes me keep thinking this is sample content...

Reply:
Agreed. I renamed it.